src/init.c

```diff
@@ -511,12 +518,28 @@ static void hyper_print_uptime(void)
 {
 	char buf[128];
 	int fd = open("/proc/uptime", O_RDONLY);
+	ssize_t bytes_read;
 
 	if (fd < 0)
 		return;
+
 	memset(buf, 0, sizeof(buf));
-	if (read(fd, buf, sizeof(buf)))
+	buf[sizeof(buf)-1] = '\0';
+
+	bytes_read = read(fd, buf, sizeof(buf)-1);
+	if (bytes_read < 0) {
+		fprintf(stdout, "reading /proc/uptime failed: %s\n",
+			strerror(errno));
+	} else if (bytes_read == 0) {
+		fprintf(stderr, "EOF reading /proc/uptime\n");
+	} else {
+		if (bytes_read > sizeof(buf)-1) {
```

Comment:
There are lots of references to `sizeof(buf)-1` in this function so I'd be inclined to define a variable or macro to avoid this duplication. Also, something like `buffer_size` is a lot easier to read than `sizeof(buf)-1`.